warehouse/integrations/middleware/sqlquerywrapper/sql.go

```diff
@@ -108,6 +109,28 @@ func (db *DB) QueryRowContext(ctx context.Context, query string, args ...interfa
 	return row
 }
 
+func (db *DB) WithTx(fn func(*Tx) error, timeout time.Duration) error {
+	tx, err := db.Begin()
+	if err != nil {
+		return fmt.Errorf("begin transaction: %w", err)
+	}
+
+	txt := txWithTimeout{
+		Tx:      tx,
+		timeout: timeout,
+	}
+
+	defer func() {
+		_ = txt.Rollback()
```

Comment:
I know that transactions cannot be rolled back after they are committed in PostgreSQL but I was wondering if that is the case with other SQL DBs as well. I'm asking because this is a generic sql wrapper that can be used with other backends as well, right?

Using always the same `err` and checking in the `defer` if `err != nil` could also be an option.